File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x = 0
    natural = 0
    polynomial = polynomial.split()
    
    for item in polynomial:
        # x의 계수인 경우
        if "x" in item:
            # 초기화 안 된 경우
            if x == 0:
                # 계수가 1인 경우
                if len(item) == 1:
                    x = 1
                else:
                    x = int(item[0:-1])
            # 초기화 된 경우
            elif len(item) == 1:
                x += 1
            else:
                x += int(item[0:-1])
        # 상수항인 경우
        elif item != "+":
            # 초기화 안 된 경우
            if natural == 0:
                natural = int(item)
            # 초기화 된 경우
            else:
                natural += int(item)
        print(x, natural)

    if x and natural:
        if x == 1:
            return "x" + " + " + str(natural)
        else:
            return str(x) + "x" + " + " + str(natural)
    if x and not natural:
        if x == 1:
            return "x"
        else:
            return str(x) + "x"
    if not x and natural:
        return str(natural)

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    var answer = '';
    polynomial = polynomial.replaceAll(" +", "").split(" ");
    let xValue = 0;
    let numValue = 0;
    polynomial.forEach((item) => {
        if (item.includes("x")) xValue += item.replace("x", "") ? Number(item.replace("x", "")) : 1;
        else numValue += Number(item);
    })
    if (xValue && numValue) return `${xValue === 1 ? "" : xValue}x + ${numValue}`;
    else if (xValue) return `${xValue === 1 ? "" : xValue}x`;
    return `${numValue}`;
}